wgpu-hal/src/dx12/device.rs

```diff
@@ -841,25 +824,27 @@ impl crate::Device for super::Device {
         //     ...
         // (bind group [0]) - Space=0
         //   View descriptor table, if any
-        //   Sampler descriptor table, if any
+        //   Sampler buffer descriptor table, if any
         //   Root descriptors (for dynamic offset buffers)
         // (bind group [1]) - Space=0
         // ...
         // (bind group [2]) - Space=0
         // Special constant buffer: Space=0
+        // Sampler descriptor tables: Space=0
+        //   SamplerState Array: Space=0, Register=0-2047
+        //   SamplerComparisonState Array: Space=0, Register=2048-4095
```

Comment:
```suggestion
        //   SamplerComparisonState Array: Space=0, Register=0-2047
```

Reply:
This is actually correct - they're different registers, but mapped to the same descriptors.

Reply:
It seemed to me that we were assigning the same registers.

Reply:
Sneaky sneaky mutable state https://github.com/gfx-rs/wgpu/pull/6766/files#diff-f6ba570302c288f76e26c7fd6839be96e600fd0368bf56d86adab2f30e70dab2R1107

Reply:
I'll clean this up a bit so it's clearer what's going on